pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -212,6 +212,11 @@ func (configgen *ConfigGeneratorImpl) buildOutboundClusters(proxy *model.Proxy,
 				// Only need the authentication MTLS mode when service is not external.
 				policy, _ := push.AuthenticationPolicyForWorkload(service, port)
 				serviceMTLSMode = authn_v1alpha1_applier.GetMutualTLSMode(policy)
+				// Conservatively set service mode to be Permissive since beta policy can use workload selector.
+				if push.AuthnBetaPolicies != nil &&
```

Comment:
I think the condition should be:

```
if policy != nil || push.AuthnBetaPolicies.MightAffectNamespaceByPeerAuthn(service.Attributes.Namespace)  {
...
}
```

as we treat no alpha nor beba policy == (beta) permissive.

Reply:
you mean ` (alphaPolicy  == nil  && betaPolicy == nil) || ( beta != nil && beta.MightAffectNamespaceByPeerAuthn) ` ?

how about i add a function into push context to collapse them, rather than inline here?

Reply:
Yeah, `(alpha)policy == nil`, sorry. You can drop the check for beta != nil as beta.MightAffectNamespaceByPeerAuthn already cover that. Anyway, +1 for a function and feel free to make the if statement more readable.

Reply:
i added a todo. moving to push context turns out introduce the same circular dependency.

we do have to check `beta != nil` just to avoid de-referencing on a nil pointer, i think.